mautrix_instagram/user.py

```diff
@@ -376,6 +378,11 @@ async def on_proxy_update(self, evt: ProxyUpdate | None = None) -> None:
         if self.command_status:
             self.command_status["api"].setup_http(self.command_status["state"].cookies.jar)
 
+    async def on_response_error(self, err: IGResponseError) -> None:
+        if isinstance(err, IGNotLoggedInError):
```

Comment:
```suggestion
        if isinstance(err, IGNotLoggedInError) and (await self.is_logged_in()):
```

@tulir this should resolve the issue with the login checks I think?

Reply:
It doesn't resolve calls like the one in `fetch_user_and_reconnect`, `_sync`, `_handle_backfill_requests_loop` and `run_with_sync_lock`, but maybe we could just remove those (and assume that the response error handler always handles it)